packages/logger/lib/src/amplify_logger.dart

```diff
@@ -0,0 +1,144 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:ansicolor/ansicolor.dart';
+import 'package:aws_common/aws_common.dart';
+
+import 'log_event.dart';
+import 'log_level.dart';
+
+class AmplifyLogger {
+  static final StreamController<LogEvent> _logStreamController =
+      StreamController<LogEvent>();
+
+  static final Sink<LogEvent> _logSink = _logStreamController.sink;
+
+  /// A map of existing logger instances.
+  static final Map<String, AmplifyLogger> _logsMap = {};
+
+  /// A stream of logs from all logger instances.
+  static final Stream<LogEvent> logStream = _logStreamController.stream;
+
+  /// Default logger instance without a namespace.
+  static get defaultLogger => getInstance();
+
+  LogLevel logLevel = LogLevel.info;
+  static LogLevel globalLogLevel = LogLevel.info;
+
+  static bool useDefaultLogHandler = true;
+
+  /// The namespace of the logger instance.
+  String get namespace {
+    return _namespace;
+  }
+
+  String _namespace;
+
+  AmplifyLogger({String namespace = 'default'}) : _namespace = namespace;
+
+  /// Log a message with an optional log level.
+  ///
+  /// The log level defaults to [LogLevel.info].
+  void log({
+    required String message,
+    LogLevel level = LogLevel.info,
+  }) {
+    if (level < logLevel) return;
+
+    final logEvent = LogEvent(
+      level: level,
+      namespace: namespace,
+      message: message,
+    );
+    _logSink.add(logEvent);
+  }
+
+  /// Log a message with a log level of [LogLevel.verbose].
+  void verbose(String message) {
+    log(message: message, level: LogLevel.verbose);
+  }
+
+  /// Log a message with a log level of [LogLevel.debug].
+  void debug(String message) {
+    log(message: message, level: LogLevel.debug);
+  }
+
+  /// Log a message with a log level of [LogLevel.info].
+  void info(String message) {
+    log(message: message, level: LogLevel.info);
+  }
+
+  /// Log a message with a log level of [LogLevel.warn].
+  void warn(String message) {
+    log(message: message, level: LogLevel.warn);
+  }
+
+  /// Log a message with a log level of [LogLevel.error].
+  void error(String message) {
+    log(message: message, level: LogLevel.error);
+  }
+
+  /// Get or create a logger instance.
+  ///
+  /// If no instance exists with the given namespace, a new
+  /// instance will be created.
+  static AmplifyLogger getInstance({String namespace = 'default'}) {
+    return _logsMap[namespace] ??= AmplifyLogger(namespace: namespace);
+  }
+
+  static AmplifyLogger getCategoryInstance({required Category category}) {
+    return getInstance(namespace: category.toString());
+  }
```

Comment:
What if users's (external developers) wanted to use the logger outside of an existing Amplify Category?

Reply:
I left this on another comment:

> Our main focus right now should be logging within our library since that is an immediate need. I would actually suggest we don't expose this initially. We should think more about the requirements for a logger that customers would use (sending logs to cloud watch, etc.) before we expose this, as those requirements might change what we expose.

Reply:
okay sounds good, see updated code